@stellar/typescript-wallet-sdk/src/walletSdk/Types/sep43.ts

```diff
@@ -0,0 +1,92 @@
+interface Error {
+  message: string; // general description message returned to the client app
+  code: number; // unique error code
+  ext?: Array<string>; // optional extended details
+}
+
+export interface Sep43Interface {
+  /**
+   * Function used to request the public key from the active account
+   *
+   * @return Promise<{ address: string } & { error?: Error }>
+   */
+  getAddress(): Promise<{ address: string } & { error?: Error }>;
+
+  /**
+   * A function to request a wallet to sign a built transaction in its XDR mode
+   *
+   * @param xdr - A Transaction or a FeeBumpTransaction
+   * @param opts - Options compatible with https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0043.md#signtransaction
+   * @param opts.networkPassphrase - The Stellar network to use when signing
+   * @param opts.address - The public key of the account that should be used to sign
+   * @param opts.path - This options is added for special cases like Hardware wallets
+   *
+   * @return Promise<{ signedTxXdr: string; signerAddress: string } & { error?: Error }>
+   */
+  signTransaction(
+    xdr: string,
+    opts?: {
+      networkPassphrase?: string;
+      address?: string;
+      path?: string;
+      submit?: boolean;
+      submitUrl?: string;
+    },
+  ): Promise<
+    { signedTxXdr: string; signerAddress?: string } & { error?: Error }
```

Comment:
It looks like maybe `signedAddress` is not optional in the spec, should it be?

Reply:
Ah good catch - it should not be optional. I'll fix this here

Reply:
nice thanks. I think this applies to the other instance of `signerAddress` as well.